plugins/inputs/vsphere/endpoint.go

```diff
@@ -818,6 +824,7 @@ func getVMs(ctx context.Context, e *Endpoint, resourceFilter *ResourceFilter) (o
 			ref:          r.ExtensibleManagedObject.Reference(),
 			parentRef:    r.Runtime.Host,
 			guest:        guest,
+			memorySizeMB: memorySizeMB,
```

Comment:
You can directly set 
```suggestion
			memorySizeMB: r.Summary.Config.MemorySizeMB,
```
here as `memorySizeMB` will be zero in case `r.Summary.Config.MemorySizeMB` is zero anyway.